src/libexpr/print.cc

```diff
@@ -90,5 +127,260 @@ printAttributeName(std::ostream & str, std::string_view name) {
+            str << ANSI_CYAN;
+        str << v.fpoint;
+        if (options.ansiColors)
+            str << ANSI_NORMAL;
+        break;
+
+    case nBool:
+        if (options.ansiColors)
+            str << ANSI_CYAN;
+        printLiteralBool(str, v.boolean);
+        if (options.ansiColors)
+            str << ANSI_NORMAL;
+        break;
+
+    case nString:
+        printLiteralString(str, v.string_view(), options.maxStringLength, options.ansiColors);
+        break;
+
+    case nPath:
+        if (options.ansiColors)
+            str << ANSI_GREEN;
+        str << v.path().to_string(); // !!! escaping?
+        if (options.ansiColors)
+            str << ANSI_NORMAL;
+        break;
+
+    case nNull:
+        if (options.ansiColors)
+            str << ANSI_CYAN;
+        str << "null";
+        if (options.ansiColors)
+            str << ANSI_NORMAL;
+        break;
+
+    case nAttrs: {
+        if (seen)
+            seen->insert(&v);
+
+        if (options.force && state.isDerivation(v)) {
+            str << "«derivation";
+            Bindings::iterator i = v.attrs->find(state.sDrvPath);
+            NixStringContext context;
+            if (i != v.attrs->end())
+                str << " " << state.store->printStorePath(state.coerceToStorePath(i->pos, *i->value, context, "while evaluating the drvPath of a derivation"));
+            str << "»";
+        }
+
+        else if (depth < options.maxDepth) {
+            str << "{ ";
+
+            typedef std::map<std::string, Value *> Sorted;
+            Sorted sorted;
+            for (auto & i : *v.attrs)
+                sorted.emplace(state.symbols[i.name], i.value);
+
+            unsigned int attrsSeen = 0;
+            for (auto & i : sorted) {
+                if (attrsSeen >= options.maxAttrs) {
+                    if (options.ansiColors)
+                        str << ANSI_FAINT;
+                    str << " «";
+                    pluralize(str, sorted.size() - attrsSeen, "attribute", "attributes");
+                    str << " elided»";
+                    if (options.ansiColors)
+                        str << ANSI_NORMAL;
+                    break;
+                }
+
+                printAttributeName(str, i.first);
+                str << " = ";
+                if (seen && seen->count(i.second)) {
+                    if (options.ansiColors)
+                        str << ANSI_MAGENTA;
+                    str << "«repeated»";
+                    if (options.ansiColors)
+                        str << ANSI_MAGENTA;
+                } else
+                    print(str, *i.second, depth + 1, seen);
+                str << "; ";
+                attrsSeen++;
+            }
+
+            str << "}";
+        } else
+            str << "{ ... }";
+
+        break;
+    }
+
+    case nList:
+        if (seen)
+            seen->insert(&v);
+
+        str << "[ ";
+        if (depth < options.maxDepth) {
+            unsigned int listItemsSeen = 0;
```

Comment:
```suggestion
            unsigned int listItemsPrinted = 0;
```

Like `attrsPrinted`, suggested in earlier comment.